src/Drupal/DrupalKernel.php

```diff
@@ -106,7 +106,11 @@ public function discoverServiceProviders()
         // necessary that the class files in these commands are available
         // in the autoloader.
 
-        // Also add Drush services from all modules
+        // Also add Drush services from all modules & themes.
+        $extensions = $this->getConfigStorage()->read('core.extension');
+        $this->moduleList += isset($extensions['theme']) ? $extensions['theme'] : [];
```

Comment:
I don't know if this is prudent. This field is not only used by Drush, but is used throughout Drupal. Adding the themes to the module list might have unintended consequences. It would probably be much better to simply build a list of themes in local variables, and then have a separate loop over `$theme_filenames` to call `addDrushServiceProvider`.

Reply:
hmmm... yes that does make sense; I wasn't considering that - I was thinking this only affects Drush. Ah well... quite easy to modify. I'll swap over to local/adjust the implementation. A bit more code/close to duplication but seems very worth it to not be altering core data without tons of testing
thanks for the review :)

Reply:
You could have less duplication by merging the module list and theme list, but I don't think you should do that either. You're not supposed to ever have a module and a theme with the same name, but I've seen that violated before, so best keep the lists separated (unless Drupal is actually preventing that now -- have not tried recently).

Reply:
Yeah I've seen that before too, which should have made me think about keeping it separate